lib/plugins/aws/package/compile/events/httpApi.js

```diff
@@ -81,6 +81,13 @@ class HttpApiEvents {
             authorizer: {
               anyOf: [
                 { type: 'string' },
+                {
+                  type: 'object',
+                  properties: {
+                    type: { const: 'aws_iam' },
+                  },
+                  additionalProperties: false,
+                },
```

Comment:
Configuring definition like that is slightly problematic as it'll trigger _vague_ mode for validation of other authorizers, as now any validation failure will now be reported simply with _"Unsupported configuration format"_ without providing a hint which exactly property is misconfigured.

It's due to fact that two versions for _object_ variant are provided, and validator won't know errors of which to report, hence it'll report just message that something is wrong.

(you may also confirm on that manually).

I think best way to recover is to use one object notation, and then inline ensure that for AWS IAM case it's just `type` property that provided, and throw meaningful error if any other properties are found

Reply:
I've adjusted the configuration so it should be able to accept all possible versions + added additional validation to ensure that we're working with valid configuration.